lib/fluent/parser.rb

```diff
@@ -548,7 +552,26 @@ def initialize
       def configure(conf)
         super
 
-        @regexp = @with_priority ? REGEXP_WITH_PRI : REGEXP
+        @time_parser_rfc3164 = @time_parser_rfc5424 = nil
+        @regexp = case @message_format
+                  when :rfc3164
+                    class << self
+                      alias_method :parse, :parse_plain
+                    end
+                    @with_priority ? REGEXP_WITH_PRI : REGEXP
+                  when :rfc5424
+                    class << self
+                      alias_method :parse, :parse_plain
+                    end
+                    REGEXP_RFC5424
```

Comment:
Yes, but time_format has default value.
So there is two approaches:

- Change time_format's default to `nil` and set each default time_format in `message_format` condition
- Use conf.has_key?('time_format')